### ISSUE TITLE
Select Layer in the MapFrame

### ISSUE THREAD

**Issue by mogikanin** [DotSpatial/DotSpatial#743]
**This issue was imported from [CodePlex](http://dotspatial.codeplex.com/workitem/63592)**

**[robertosalemi](http://www.codeplex.com/site/users/view/robertosalemi)** wrote 2015-08-12 at 19:09
Hi, 

I would select a Layer in the MapFrame, so I wrote:

<pre><code>int indexLayerPdl = AppManager.Map.Layers.IndexOf(layerAdded);
            AppManager.Map.Layers.SelectLayer(indexLayerPdl);</code></pre>
where layerAdded is a IMapFeatureLayer created before. 

The value of indexLayerPdl is 4, ma in the MapFrame the layer is not selected. 

Why? 

Thanks!

**[robertosalemi](http://www.codeplex.com/site/users/view/robertosalemi)** wrote 2015-10-09 at 11:06
Help me!

**Comment by mogikanin** [comment]
in DS <2.0 code to select layer is: 
```
var selectedIndex = 0;

((Legend)App.Map.Legend).ClearSelection();
App.Map.Layers[selectedIndex].IsSelected = true;
App.Map.Layers.SelectedLayer = App.Map.Layers[selectedIndex];
App.Map.Legend.RefreshNodes();
```

But this logic is not obvious, I'll plan to fix it in 2.0.

**Comment by mogikanin** [comment]
In 2.0 enough to do:
```
var selectedIndex = 0;
((Legend)App.Map.Legend).ClearSelection();

App.Map.Layers[selectedIndex].IsSelected = true;
// OR
App.Map.Layers.SelectedLayer = App.Map.Layers[selectedIndex];
// OR
App.Map.Layers.SelectLayer(selectedIndex);
```
